### PR TITLE
Added exception for persisting autoscaler policy in registry manager

### PR DESCRIPTION
**Pull request by Vishanth**
Added registry exception for persisting autoscaler policy in registry manager
Also ensured the autoscaler policy is not added to the information model in policy manager, if the adding of the autoscaler policy is unsuccessful

### CODE REVIEW

**Comment by imesh** [comment]
Thanks for the fix @Vishanth! I have now merged this to master branch.

File: components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java [hunk 1]
```diff
@@ -117,11 +117,17 @@ private void persist(Object dataObj, String resourcePath) throws AutoScalerExcep
...
-                    autoscalePolicy.getId(), autoscalePolicy.getDisplayName(), autoscalePolicy.getDescription()));
+        try {
+            String resourcePath = AutoscalerConstants.AUTOSCALER_RESOURCE +
+                    AutoscalerConstants.AS_POLICY_RESOURCE + "/" + autoscalePolicy.getId();
+            persist(autoscalePolicy, resourcePath);
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Autoscaler policy written to registry: [id] %s [name] %s [description] %s",
+                        autoscalePolicy.getId(), autoscalePolicy.getDisplayName(), autoscalePolicy.getDescription()));
+            }
+        } catch (Exception e) {
```
Comment (prabathabey) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java]: If catching generic exceptions was done intentionally, maybe it's good to add a comment explaining why it was done.

File: components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/pojo/policy/PolicyManager.java [hunk 2]
```diff
@@ -71,11 +71,11 @@ public boolean addAutoscalePolicy(AutoscalePolicy policy) throws InvalidPolicyEx
         if (log.isInfoEnabled()) {
             log.info(String.format("Adding autoscaling policy: [id] %s", policy.getId()));
         }
-        if(StringUtils.isEmpty(policy.getId())){
+        if (StringUtils.isEmpty(policy.getId())) {
             throw new AutoScalerException("Autoscaling policy id cannot be empty");
```
Comment (muthulee) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/pojo/policy/PolicyManager.java]: More informative logs. What policy is empty?

File: components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/pojo/policy/PolicyManager.java [hunk 3]
```diff
@@ -71,11 +71,11 @@ public boolean addAutoscalePolicy(AutoscalePolicy policy) throws InvalidPolicyEx
...
         }
-        this.addASPolicyToInformationModel(policy);
         RegistryManager.getInstance().persistAutoscalerPolicy(policy);
+        this.addASPolicyToInformationModel(policy);
         if (log.isInfoEnabled()) {
             log.info(String.format("Autoscaling policy is added successfully: [id] %s", policy.getId()));
```
Comment (muthulee) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/pojo/policy/PolicyManager.java]: Why use format for one string?

Comment (Vishanth) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/pojo/policy/PolicyManager.java]: For consistency we have been using the string format in all our loggings. But true it is not much effective for a single string.

File: components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java (hunk 1 above)
Comment (Vishanth) [inline components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/registry/RegistryManager.java]: Noted. Will do.

**Pull request closed without merging**